plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.ts

```diff
@@ -508,6 +518,61 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
     };
   }
 
+  async listAncestors(
+    txOpaque: Transaction,
+    options: ListAncestorsOptions,
+  ): Promise<ListAncestorsResult> {
+    const tx = txOpaque as Knex.Transaction;
+    const { entityRef } = options;
+    const entityRefs = new Array<string>();
+
+    let currentRef = entityRef;
+    let depth = 0;
+    for (;;) {
```

Comment:
matter of taste, but this would have fit great as a `for (let depth = 1; depth <= MAX_ANCESTOR_DEPTH; depth += 1)` and putting the `throw` under the loop if we fall through it

Reply:
I like it a lot! Fixed 👍